src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -385,7 +385,14 @@ private void AddCredentialToCache(Uri serverUri, ulong authAvail, NetworkCredent
                     if ((authAvail & s_authSchemePriorityOrder[i]) != 0)
                     {
                         Debug.Assert(_credentialCache != null, "Expected non-null credential cache");
-                        _credentialCache.Add(serverUri, s_authenticationSchemes[i], nc);
+                        try
+                        {
+                            _credentialCache.Add(serverUri, s_authenticationSchemes[i], nc);
+                        }
+                        catch(ArgumentException)
+                        {
+                            //Ignore the case of key already present
```

Comment:
Actually, I do not know how common this case could be, but
1. The CredentialCache does not provide an ```Exists``` method to see if the given pair is already present
2. I copied the same strategy as used in [WinHttpHandler](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs#L1063)